packages/data-point-service/lib/cache-middleware.js

```diff
@@ -244,40 +272,63 @@ function before (service, ctx, next) {
  * @param {Function} next Middleware callback
  */
 function after (service, ctx, next) {
-  const cache = getCacheParams(ctx.context.params)
+  const { generateKey, setStaleWhileRevalidateEntry } = module.exports
 
-  if (!cache.ttl) {
+  const cache = EntityCacheParams.getCacheParams(ctx.context.params)
+
+  // do nothing if:
+  // - cache is not configured, or
+  // - its at the process of revalidating, then lets skip any further calls
+  if (!cache.ttl || ctx.locals.revalidatingCache) {
     // do nothing
-    return next()
+    next()
+    return false
   }
 
   // from here below ttl is assumed to be true
   const entryKey = generateKey(cache.cacheKey, ctx)
 
-  if (cache.useStaleWhileRevalidate) {
-    // if its at the process of revalidating, then lets skip any further calls
-    if (ctx.locals.revalidatingCache) {
-      return next()
-    }
+  let resolution
 
+  if (cache.useStaleWhileRevalidate) {
     // adds (or updates) the stale cache entry with the latest value
-    return module.exports
-      .setStaleWhileRevalidateEntry(service, entryKey, ctx.value, cache)
-      .then(() => {
-        next()
-      })
+    resolution = setStaleWhileRevalidateEntry(
+      service,
+      entryKey,
+      ctx.value,
+      cache
+    ).then(() => {
+      next()
```

Comment:
this should be removed, because next is being called on line 314 also

Reply:
thank you!!! thank you!